src/components/Permissions/DataList.jsx

```diff
@@ -1,15 +1,114 @@
+/* eslint-disable no-unsafe-optional-chaining */
 import React from 'react'
+import { Link } from 'react-router-dom'
 import Page from 'components/Page'
 import PageTitle from 'components/PageTitle'
-import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import List from 'cozy-ui/transpiled/react/MuiCozyTheme/List'
+import ListItem from 'cozy-ui/transpiled/react/MuiCozyTheme/ListItem'
+import ListItemText from 'cozy-ui/transpiled/react/ListItemText'
+import ListItemIcon, {
+  mediumSize
+} from 'cozy-ui/transpiled/react/MuiCozyTheme/ListItemIcon'
+import Divider from 'cozy-ui/transpiled/react/MuiCozyTheme/Divider'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
+import { APPS_DOCTYPE, KONNECTORS_DOCTYPE } from 'doctypes'
+import withAllLocales from '../../lib/withAllLocales'
+import { getPermissionIconName } from '../../containers/helpers/permissionsHelper'
+import CozyClient, {
+  Q,
+  useQuery,
+  isQueryLoading,
+  hasQueryBeenLoaded
+} from 'cozy-client'
 
-const DataList = () => {
-  const { t } = useI18n()
+const DataList = ({ t }) => {
+  const THIRTY_SECONDS = 30 * 1000
```

Comment:
let's define the constant outside the component 🙏

Reply:
Is it relevant to put this constant in the Doctypes folder?

Reply:
I am suggesting to set it 2 lines above